.github/workflows/data-prepper-performance-test-compile-check.yml

```diff
@@ -0,0 +1,30 @@
+# This workflow will build a Java project with Gradle
+# For more information see: https://help.github.com/actions/language-and-framework-guides/building-and-testing-java-with-gradle
+
+name: Data Prepper Performance Tests Compile
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+  workflow_dispatch:
+
+jobs:
+  build:
+    strategy:
+      matrix:
+        java: [14]
+
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Set up JDK ${{ matrix.java }}
+        uses: actions/setup-java@v1
+        with:
+          java-version: ${{ matrix.java }}
+      - name: Checkout Data-Prepper
+        uses: actions/checkout@v2
+      - name: Grant execute permission for gradlew
+        run: chmod +x gradlew
```

Comment:
Is `gradlew` not already executable?

Reply:
It appears our current GHAs do this. We can keep this here for now. Maybe we can experiment with it later.

Reply:
I noticed other workflows have this same step. I know this can be a problem with files committed from windows. I can remove the step if this isn't a concern.

Reply:
I think we should move forward and look into it later.